web/src/NetworkManager.jsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useReducer, useState } from "react";
+import { useInstallerClient } from "./context/installer";
+import Popup from "./Popup";
+import { Button, Text } from "@patternfly/react-core";
+
+const initIpData = {
+  addresses: [],
+  hostname: ""
+};
+
+const reducer = (state, action) => {
+  const data = action.payload;
+
+  switch (action.type) {
+    case "READ": {
+      return {
+        ...state,
+        ...data
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+function formatIp(address, prefix) {
+  return address + "/" + prefix;
+}
+
+function LoadNmConfig() {
+  const client = useInstallerClient();
+  const [state, dispatch] = useReducer(reducer, initIpData);
+
+  useEffect(() => {
+    const config = async () => {
+      const data = await client.network.config();
+
+      dispatch({
+        type: "READ",
+        payload: data
+      });
+    };
+
+    config();
+  }, [client.network]);
+
+  return state;
+}
+
+export default function DetailsPopup() {
```

Comment:
NP: I'd keep export names _in sync_ with file name. As I already said, actually it _does not matter_ which name you use for the default export since you can import it using the name of your preference, but just to follow the same style as the other components.